processor/deltatocumulativeprocessor/internal/streams/limit.go

```diff
@@ -11,20 +11,31 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/streams"
 )
 
-func Limit[T any](m Map[T], max int) Map[T] {
+func Limit[T any](m Map[T], max int) LimitMap[T] {
 	return LimitMap[T]{Map: m, Max: max}
 }
 
 type LimitMap[T any] struct {
 	Max int
+
+	Evictor streams.Evictor
 	streams.Map[T]
 }
 
 func (m LimitMap[T]) Store(id identity.Stream, v T) error {
-	if m.Map.Len() >= m.Max {
-		return ErrLimit(m.Max)
+	if m.Map.Len() < m.Max {
+		return m.Map.Store(id, v)
+	}
+
+	errl := ErrLimit(m.Max)
+	if m.Evictor != nil {
+		gone := m.Evictor.Evict()
+		if err := m.Map.Store(id, v); err != nil {
+			return err
+		}
+		return ErrEvicted{ErrLimit: errl, id: gone}
```

Comment:
at this point they are errors imo, the consumer (processor) should decide what to do with them.
when https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31363 is merged, those will turn into metrics.
Until then I think they are best treated as errors

Reply:
Sure. But I guess my point is that eviction itself isn't an error. It's just something that happens. IE, eviction should not increment the `metrics_processed{error=true}` metric (I don't recall the correct name :P ).

Perhaps the eviction yes/no value should be returned as an additional return, instead of being convolved with the error status. Thoughts?

Reply:
I agree eviction should not increment that specific counter, but rather be `streams_evicted`, so you can easily alert on that. Imo eviction should never be part of normal operations, as it signals a fairly serious capacity issue and thus is very close to an error scenario.

I think it's a not uncommon thing in Go to use the Error for non-fatal occurrences, as that's the whole point of having a value-based, easy to manipulate error system.